lib/ansible/galaxy/collection.py

```diff
@@ -1140,6 +1141,9 @@ def find_existing_collections(path, fallback_metadata=False):
         for b_collection in os.listdir(b_namespace_path):
             b_collection_path = os.path.join(b_namespace_path, b_collection)
             if os.path.isdir(b_collection_path):
+                if any(b_collection_path.endswith(b_path) for b_path in b_ignore_dirs):
```

Comment:
Do you have a specific reproducer for the bug this fixes? It would be good to have tests. Since `find_existing_collections` is used by multiple subcommands I'm worried about this causing unexpected issues for some (like `install`; if it looks like the directory doesn't exist, running `ansible-galaxy collection install ` could overwrite the ignored directory without requiring `--force` or indicating something was just nuked).

Reply:
I agree with @s-hertel: it seems like this could result in a valid collection being ignore if it has a path ending with one of the ignored directories.